src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSettings.java

```diff
@@ -94,6 +87,76 @@ public boolean equals(Object o) {
     }
   }
 
+  public record VectorConfig(
+      boolean vectorEnabled,
+      int vectorSize,
+      SimilarityFunction similarityFunction,
+      VectorizeConfig vectorizeConfig) {
+
+    public static VectorConfig notEnabledVectorConfig() {
+      return new VectorConfig(false, -1, null, null);
+    }
+
+    public static VectorConfig fromJson(JsonNode jsonNode) {
+      // dimension, similarityFunction, must exist
+      int dimension = jsonNode.get("dimension").asInt();
+      SimilarityFunction similarityFunction =
+          SimilarityFunction.fromString(jsonNode.get("metric").asText());
+
+      VectorizeConfig vectorizeConfig = null;
+      // construct vectorizeConfig
+      if (jsonNode.has("service")) {
+        JsonNode vectorizeServiceNode = jsonNode.get("service");
+        // provider, model_name, authentication, must exist
+        String provider = vectorizeServiceNode.get("provider").asText();
+        String modelName = vectorizeServiceNode.get("model_name").asText();
+        // construct VectorizeConfig.VectorizeServiceAuthentication
+        JsonNode vectorizeServiceAuthenticationNode = vectorizeServiceNode.get("authentication");
+        Set<AuthenticationType> authenticationTypeSet = new HashSet<>();
+        vectorizeServiceAuthenticationNode
+            .get("type")
+            .forEach(
+                node -> authenticationTypeSet.add(AuthenticationType.fromString(node.asText())));
+        // when stored_secrets authentication is used must be provided with the name of a
+        // pre-registered secret.
+        String authenticationSecretName =
+            vectorizeServiceAuthenticationNode.has("secret_name")
```

Comment:
In this specific case, just call `get("secret_name")` since that returns `null`; no need to call `has()` first.

Reply:
Will keep this one, since I need to call asText() later, so want to avoid null pointer here.